flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierAligner.java

```diff
@@ -157,7 +162,8 @@ else if (barrierId > currentCheckpointId) {
 		else {
```

Comment:
we can also reduce this path for simple by https://github.com/apache/flink/pull/11351/files#r397598435